### PR TITLE
Add Prometheus counter for simulation staleness

### PR DESCRIPTION
**Pull request by apmasell**
Counts the number of simulation requests that request live (refreshed) or stale (not refreshed).

JIRA Ticket: 

- [X] Updates Changelog
- [X] Updates developer documentation

### CODE REVIEW

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java [hunk 1]
```diff
@@ -73,6 +74,13 @@
 import org.objectweb.asm.commons.Method;
 
 public abstract class BaseSimulateRequest {
+  private static final Counter CACHE_REFRESH =
+      Counter.build(
+              "shesmu_simulate_refresh",
+              "The number of simulation requests and whether the cache refresh should be"
```
Comment (avarsava) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java:80]: 'should be triggered' suggests an action needed on the part of anyone looking at the metrics imo. 'and the cache freshness requested'? i don't know if there's particular tense conventions for metric descriptions

**Review by avarsava — APPROVED** [review]

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java (hunk 1 above)
Comment (apmasell) [reply shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java:80]: Okay. I am bad at writing these.

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java [hunk 2]
```diff
@@ -533,10 +541,12 @@ public void function(
                         .collect(
                             Collectors.toMap(
                                 Function.identity(),
-                                name ->
-                                    inputSource
-                                        .fetch(name, readStale)
-                                        .collect(Collectors.toList())));
+                                name -> {
+                                  CACHE_REFRESH.labels(readStale ? "stale" : "live", name).inc();
```
Comment (callunity) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java:545]: It might be clearer to use the same labels as are in the UI: `cached` & `fresh`. I'm happy to make this change in another PR if you'd rather not.

**Review by callunity — APPROVED** [review]

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java (hunk 2 above)
Comment (apmasell) [reply shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/BaseSimulateRequest.java]: Fixed.
